app/services/hyrax/custom_queries/find_by_date_range.rb

```diff
@@ -26,29 +26,10 @@ def initialize(query_service:)
       # @param start_datetime [DateTime]
       # @param end_datetime [DateTime]
       def find_by_date_range(start_datetime:, end_datetime: nil, models: nil)
-        end_datetime = 1.second.since(Time.zone.now) if end_datetime.blank?
-        if models.present?
-          query_service.run_query(find_models_by_date_range_query, start_datetime.to_s, end_datetime.to_s, models)
-        else
-          query_service.run_query(find_by_date_range_query, start_datetime.to_s, end_datetime.to_s)
-        end
-      end
-
-      def find_models_by_date_range_query
-        <<-SQL
-          SELECT * FROM orm_resources
-          WHERE created_at >= ?
-          AND created_at <= ?
-          AND internal_resource IN (?);
-        SQL
-      end
-
-      def find_by_date_range_query
-        <<-SQL
-          SELECT * FROM orm_resources
-          WHERE created_at >= ?
-          AND created_at <= ?;
-        SQL
+        end_range = end_datetime.blank? ? '*' : end_datetime.utc.xmlschema
+        args = "system_create_dtsi:[#{start_datetime.utc.xmlschema} TO #{end_range}]"
+        args += " AND has_model_ssim: (#{models.map { |m| "\"#{m}\"" }.join(' OR ')})" unless models.empty?
+        Hyrax::SolrService.query(args)
```

Comment:
does this change the flavor of thing returned? instead of `models [Array]` is it now `SolrHit [Array]`?

Reply:
Yeah, this should be changed to return a model.

Reply:
I'm concerned performance of these statistics related custom queries is going to be terrible on production sized data sets.